### PR TITLE
Jmurray/file attachments fix

### PR DESCRIPTION
**Pull request by jmurray-d2l**
the attachementsHref was not being updated correctly.  As suggested by Alex I changed the pattern to grab the href from the entity when needed instead of caching it, I applied this pattern to the other members that were also being cached for consistency and to avoid more state bugs

@vix633 richTextEditorDisabled doesn't look like it is being used?  I removed it, let me know if thats a problem

### CODE REVIEW

File: components/consistent-evaluation-page.js [hunk 1]
```diff
@@ -48,14 +49,6 @@ export default class ConsistentEvaluationPage extends SkeletonMixin(LocalizeCons
 				attribute: 'special-access-href',
 				type: String
 			},
-			richTextEditorDisabled: {
```
Comment (vix633) [inline components/consistent-evaluation-page.js:51]: Just test to make sure the rich text editor still works.

File: components/consistent-evaluation-page.js [hunk 2]
```diff
@@ -230,7 +220,7 @@ export default class ConsistentEvaluationPage extends SkeletonMixin(LocalizeCons
 
 	get _feedbackText() {
 		if (this._feedbackEntity && this._feedbackEntity.properties) {
-			return this._feedbackEntity.properties.text || '';
+			return this._feedbackEntity.properties.html || '';
```
Comment (jmurray-d2l) [inline components/consistent-evaluation-page.js:223]: This change is unrelated, thought I would just sneak it in here.  Any styling that was applied in the feedback box was not being shown upon reload.  The html property has the correct styling

File: components/right-panel/consistent-evaluation-feedback-presentational.js [hunk 3]
```diff
@@ -112,7 +112,7 @@ class ConsistentEvaluationFeedbackPresentational extends LocalizeConsistentEvalu
 	}
 
 	render() {
-		if (this.href && this.token) {
+		if (this.href && this.token && this.richTextEditorConfig) {
```
Comment (jmurray-d2l) [inline components/right-panel/consistent-evaluation-feedback-presentational.js:115]: This just stops an initial error from appearing in the console, no functional difference

File: components/consistent-evaluation-page.js [hunk 4]
```diff
@@ -274,10 +264,6 @@ export default class ConsistentEvaluationPage extends SkeletonMixin(LocalizeCons
 		const bypassCache = true;
 		this.evaluationEntity = await this._controller.fetchEvaluationEntity(bypassCache);
 		this.evaluationState = this.evaluationEntity.properties.state;
-		this.allowEvaluationWrite = this._controller.userHasWritePermission(this.evaluationEntity);
```
Comment (AlexBedley) [inline components/consistent-evaluation-page.js:277]: hmm weird that these weren't properties. oh well

File: components/consistent-evaluation-page.js [hunk 5]
```diff
@@ -557,6 +541,44 @@ export default class ConsistentEvaluationPage extends SkeletonMixin(LocalizeCons
 		}));
 	}
 
+	_getAttachmentsLink() {
```
Comment (AlexBedley) [inline components/consistent-evaluation-page.js:544]: might make more sense for these 4 functions to live in a helper class for unit-testability

File: components/consistent-evaluation-page.js [hunk 6]
```diff
@@ -557,6 +541,44 @@ export default class ConsistentEvaluationPage extends SkeletonMixin(LocalizeCons
...
+		if (!this.evaluationEntity || !this.evaluationEntity.hasLinkByRel(attachmentsRel)) {
+			return null;
```
Comment (AlexBedley) [inline components/consistent-evaluation-page.js:546]: what does `IfDefined` do with `null`?

File: components/consistent-evaluation-page.js [hunk 7]
```diff
@@ -596,19 +618,18 @@ export default class ConsistentEvaluationPage extends SkeletonMixin(LocalizeCons
 						rubric-assessment-href=${ifDefined(this.rubricAssessmentHref)}
 						outcomes-href=${ifDefined(this.outcomesHref)}
 						coa-eval-override-href=${ifDefined(this.coaDemonstrationHref)}
-						attachments-href=${ifDefined(this.attachmentsHref)}
-						.richTextEditorConfig=${this.richtextEditorConfig}
+						attachments-href=${ifDefined(this._getAttachmentsLink())}
```
Comment (AlexBedley) [inline components/consistent-evaluation-page.js:621]: I wonder if lit is smart enough to know to re-render if any dependencies in these functions change. probably doesn't matter for this PR (since we use `evaluationEntity` elsewhere in `render`)

**Review by AlexBedley — APPROVED** [review]

File: components/consistent-evaluation-page.js (hunk 6 above)
Comment (dbatiste) [reply components/consistent-evaluation-page.js:546]: The `ifDefined` directive only looks for `undefined`.  It treats `null` as a valid value.

Comment (jmurray-d2l) [reply components/consistent-evaluation-page.js:546]: I'll update to return undefined
